app/src/components/results/layoutComposer.test.js

```diff
@@ -0,0 +1,375 @@
+        cardinality: 'one'
+      },
+      {
+        fieldname: 'resourceversion',
+        fieldtype: 'string',
+        mandatory: true,
+        cardinality: 'one'
+      },
+      {
+        fieldname: 'nodename',
+        fieldtype: 'relationship',
+        mandatory: true,
+        refdatatype: 'node',
+        cardinality: 'one'
+      },
+      {
+        fieldname: 'ownertype',
+        fieldtype: 'string',
+        mandatory: false,
+        cardinality: 'one'
+      },
+      {
+        fieldname: 'creationtime',
+        fieldtype: 'string',
+        mandatory: true,
+        cardinality: 'one'
+      },
+      {
+        fieldname: 'k8sobj',
+        fieldtype: 'true',
+        mandatory: true,
+        cardinality: 'one'
+      },
+      {
+        fieldname: 'objtype',
+        fieldtype: 'string',
+        mandatory: true,
+        cardinality: 'one'
+      },
+      {
+        fieldname: 'volumes',
+        fieldtype: 'json',
+        mandatory: false,
+        cardinality: 'many'
+      },
+      {
+        fieldname: 'owner',
+        fieldtype: 'relationship',
+        mandatory: false,
+        refdatatype: 'replicaset,daemonset,statefulset',
+        cardinality: 'one'
+      }
+    ]
+  }
+};
+const dataItem = {
+  availablereplicas: '1',
+  creationtime: '2019-01-18T01:18:36Z',
+  k8sobj: 'k8sobj',
+  labels:
+    '{"app":"foremast-service","applications.argoproj.io/app-name":"foremast-brain-usw2-ppd"}',
+  name: 'foremast-service',
+  numreplicas: '1',
+  objtype: 'deployment',
+  resourceid:
+    'deployment:paas-preprod-west2.cluster.k8s.local:dev-containers-foremast-brain-usw2-ppd-ppd:foremast-service',
+  resourceversion: '90347576',
+  strategy: 'RollingUpdate',
+  uid: '0x10f0f9',
+  '~owner': [
+    {
+      creationtime: '2019-01-18T17:41:01Z',
+      k8sobj: 'k8sobj',
+      labels:
+        '{"app":"foremast-service","applications.argoproj.io/app-name":"foremast-brain-usw2-ppd","pod-template-hash":"3387190116"}',
+      name: 'foremast-service-77dc5f455b',
+      numreplicas: '1',
+      objtype: 'replicaset',
+      resourceid:
+        'replicaset:paas-preprod-west2.cluster.k8s.local:dev-containers-foremast-brain-usw2-ppd-ppd:foremast-service-77dc5f455b',
+      resourceversion: '90347574',
+      uid: '0x120272',
+      '~owner': [
+        {
+          containers:
+            '[{"name":"service","image":"docker.artifactory.a.intuit.com/foremast/foremast-service:0.0.5","ports":[{"name":"http","containerPort":8099,"protocol":"TCP"}],"env":[{"name":"ELASTIC_URL","value":"http://elasticsearch-discovery.dev-containers-foremast-brain-usw2-ppd-ppd.svc.cluster.local:9200/"}],"resources":{"limits":{"cpu":"100m","memory":"30Mi"},"requests":{"cpu":"100m","memory":"20Mi"}},"volumeMounts":[{"name":"default-token-zxbmj","readOnly":true,"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount"}],"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File","imagePullPolicy":"Always"}]',
```

Comment:
may want to make URIs in test data more generic

Reply:
agree. will use some dummy ones.